src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -109,14 +117,124 @@ protected Object readResolve() throws ObjectStreamException {
      * <p>It will be false for a node which still has active children, like a step with a running body.
      * It will also be false for something that has finished but is pending child node creation,
      * such as a completed fork branch which is waiting for the join node to be created.
-     * <p>This can only go from true to false and is a shortcut for {@link FlowExecution#isCurrentHead}.
+     * <p>This can only go from true to false.
+     * @deprecated Usually {@link #isActive} is what you want. If you really wanted the original behavior, use {@link FlowExecution#isCurrentHead}.
      */
-    @Exported
+    @Deprecated
     public final boolean isRunning() {
         return getExecution().isCurrentHead(this);
     }
 
     /**
+     * Checks whether a node is still part of the active part of the graph.
+     * Unlike {@link #isRunning}, this behaves intuitively for a {@link BlockStartNode}:
+     * it will be considered active until the {@link BlockEndNode} is added.
+     */
+    @Exported(name="running")
+    public final boolean isActive() {
```

Comment:
Specifically:  I think the most common case is a node either being in the current heads (active=true) or NOT being a block start/end node AND not being in the current heads (active=false).  I think putting that conditional first may make the hottest path perform well.